draft/ZEP0010.md

```diff
@@ -0,0 +1,413 @@
+---
+layout: default
+title: ZEP010
+description: This ZEP proposes a new generic extensions field.
+parent: draft ZEPs
+nav_order: 10
+---
+
+# ZEP 10 — Zarr Generic Extensions
+
+Authors:
+
+- [Norman Rzepka](https://github.com/normanrz), scalable minds
+- [Josh Moore](https://github.com/joshmoore), German BioImaging
+
+Status: Draft
+
+Type: Specification
+
+Created: 2025-05-12
+
+## Abstract
+
+This proposal defines a new generic extension point, ``extensions``, to be
+included in the metadata of Zarr v3 arrays and groups. The ``extensions`` field
+provides a consistent mechanism for attaching additional metadata that does not
+fit into existing extension points defined by the core specification. Extension
+entries within this field follow the naming and structure rules established in
+ZEP0009. This mechanism enables third parties to define and share metadata
+extensions without requiring changes to the core specification or introducing
+new top-level keys.
+
+## Introduction
+
+Zarr specification version 3 currently defines four extension points, each
+associated with a specific (array) metadata field. Additional extension points
+may be added by future ZEPs. Until that time, however, third-parties may want
+to add arbitrary extension objects to either arrays or groups. This proposal
+introduces a generic ``extensions`` field that serves as a container for such a
+list of extensions.
+
+## Proposal
+
+To provide for more flexible, immediate, and de-centralized use cases, we
+propose to add a generic extension point ```extensions`` on
+both arrays and groups into which extensions MAY be added.
+
+The ``extensions`` field if present MUST contain an array of extension
+definitions. The contained array MUST either have one or more extensions or the
+object MUST be omitted entirely.
+
+Further details on the specification changes can be found in
+<https://github.com/zarr-developers/zarr-specs/pull/344>.
+
+### Definition and naming
+
+Each extension object will follow the rules laid out in the "Zarr extensions"
+section of the v3 specification.
+
+### Processing
+
+In a manner similar to OpenGL’s extension mechanism, where implementations must
+explicitly advertise and support extensions they recognize, Zarr implementers
+are expected to inspect the extensions array and determine whether each listed
+extension is supported. If an extension includes ``"must_understand": true``
```

Comment:
I don't follow the connection to opengl here. How should zarr implementations explicitly advertise the extensions they support? How could implementations recognize, but not support an extension? 

Also, I have no idea how opengl's extensions work. People like might might benefit from a short description or link, something to make the opengl reference persuasive / illustrative.

Reply:
To me the analogy with OpenGL was useful, but it might be confusing to others and not strictly necessary to understand our proposal.